### PR TITLE
HAL-1504: Add tests for new modcluster resources

### PR DESCRIPTION
**Pull request by hpehl**
https://issues.jboss.org/browse/HAL-1504

### CODE REVIEW

File: tests/basic/src/test/java/org/jboss/hal/testsuite/test/runtime/microprofile/health/MicroProfileHealthCheckTest.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+import static org.jboss.hal.testsuite.fragment.finder.FinderFragment.runtimeSubsystemPath;
+import static org.junit.Assert.assertTrue;
+
+@RunWith(Arquillian.class)
+@Category(Standalone.class)
+public class MicroProfileHealthCheckTest {
+
+    private static final OnlineManagementClient client = ManagementClientProvider.createOnlineManagementClient();
+    private static final ServerEnvironmentUtils serverEnvironmentUtils = new ServerEnvironmentUtils(client);
+
```
Comment (spriadka) [inline tests/basic/src/test/java/org/jboss/hal/testsuite/test/runtime/microprofile/health/MicroProfileHealthCheckTest.java]: Please close the client after the test execution, e.g by calling `client.close()` in the `@AfterClass` method.

File: tests/basic/src/test/java/org/jboss/hal/testsuite/test/configuration/modcluster/CustomLoadMetricTest.java [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+import static org.jboss.hal.dmr.ModelDescriptionConstants.*;
+import static org.jboss.hal.testsuite.test.configuration.modcluster.ModclusterFixtures.*;
+import static org.jboss.hal.testsuite.test.configuration.modcluster.ModclusterFixtures.CLASS_NAME;
+
+@RunWith(Arquillian.class)
+public class CustomLoadMetricTest {
+
+    private static final OnlineManagementClient client = ManagementClientProvider.createOnlineManagementClient();
+    private static final Operations operations = new Operations(client);
+
```
Comment (spriadka) [inline tests/basic/src/test/java/org/jboss/hal/testsuite/test/configuration/modcluster/CustomLoadMetricTest.java]: Please close the client after test execution

File: tests/basic/src/test/java/org/jboss/hal/testsuite/test/runtime/microprofile/health/MicroProfileHealthCheckTest.java (hunk 1 above)
Comment (hpehl) [reply tests/basic/src/test/java/org/jboss/hal/testsuite/test/runtime/microprofile/health/MicroProfileHealthCheckTest.java]: Added code to close client in `@AfterClass` method.

File: tests/basic/src/test/java/org/jboss/hal/testsuite/test/configuration/modcluster/CustomLoadMetricTest.java (hunk 2 above)
Comment (hpehl) [reply tests/basic/src/test/java/org/jboss/hal/testsuite/test/configuration/modcluster/CustomLoadMetricTest.java]: Added code to close client in `@AfterClass` method.

File: tests/basic/src/test/java/org/jboss/hal/testsuite/test/configuration/modcluster/CustomLoadMetricTest.java [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+        proxyAdd.add(proxyAddress(PROXY_UPDATE), Values.of(CONNECTOR, DEFAULT));
+        proxyAdd.add(loadProviderDynamicAddress(PROXY_UPDATE));
+        operations.batch(proxyAdd);
+        operations.add(customLoadMetricAddress(PROXY_UPDATE, CUSTOM_LOAD_METRIC_DELETE), Values.of(CLASS, CLASS_NAME));
+        operations.add(customLoadMetricAddress(PROXY_UPDATE, CUSTOM_LOAD_METRIC_UPDATE), Values.of(CLASS, CLASS_NAME));
+    }
+
+    @AfterClass
+    public static void afterClass() throws Exception {
+        operations.remove(proxyAddress(PROXY_UPDATE));
```
[the remark this answers is not in the document]
Comment (hpehl) [reply tests/basic/src/test/java/org/jboss/hal/testsuite/test/configuration/modcluster/CustomLoadMetricTest.java]: Replaced with `removeIfExists()`

File: tests/basic/src/test/java/org/jboss/hal/testsuite/test/configuration/modcluster/ModclusterLoadProviderSimpleTest.java [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+
+import static org.jboss.hal.dmr.ModelDescriptionConstants.DEFAULT;
+import static org.jboss.hal.dmr.ModelDescriptionConstants.LISTENER;
+import static org.jboss.hal.dmr.ModelDescriptionConstants.NAME;
+import static org.jboss.hal.testsuite.test.configuration.modcluster.ModclusterFixtures.*;
+import static org.junit.Assert.assertTrue;
+import static org.junit.runners.MethodSorters.NAME_ASCENDING;
+
+@RunWith(Arquillian.class)
+@FixMethodOrder(NAME_ASCENDING)
```
[the remark this answers is not in the document]
Comment (hpehl) [reply tests/basic/src/test/java/org/jboss/hal/testsuite/test/configuration/modcluster/ModclusterLoadProviderSimpleTest.java]: Added code to close client in `@AfterClass` method.

**Comment by hpehl** [comment]
@spriadka should be ready to merge now.

**Comment by spriadka** [comment]
@hpehl, `ModclusterLoadProviderDynamicTest` and `ModclusterLoadProviderSimpleTest` are still using the @FixedMethodOrder annotation to run test methods in some order. This still goes against the basic testing principle of running tests independently. When the execution of a single test fails, the rest of the depending tests will fail as well. Also, I suppose to keep the test method execution in alphabetical order the names of test methods that were used are misleading in this case such as `_01noSimpleProvider()`

**Comment by hpehl** [comment]
@spriadka Please review and merge.

**Pull request merged**